lib/grape/validations/validators/default.rb

```diff
@@ -7,7 +7,24 @@ def initialize(attrs, options, required, scope)
       end
 
       def validate_param!(attr_name, params)
-        params[attr_name] = @default.is_a?(Proc) ? @default.call : @default unless params.key?(attr_name)
+        return if params.key? attr_name
+        params[attr_name] = if @default.is_a? Proc
+                              @default.call
+                            elsif @default.nil? ||
+                                  @default == true ||
+                                  @default == false ||
+                                  @default.is_a?(Symbol) ||
+                                  @default.is_a?(Numeric)
+                              # We know clone won't work in these cases, so just use the object.
+                              @default
+                            else
+                              # for the rest, try to clone but use the object if that fails.
+                              begin
+                                @default.clone
+                              rescue TypeError
```

Comment:
If the default value is `frozen` to begin with, we could safely continue to use it as-is since there's no danger of side-effects via modification (which is the bug we're fixing). Doing that would not be a change from how things work today, so it won't surprise anyone. 

And, if the user really wants the default to be a `dup` of a `frozen` object, they have the ability to do that by using a `Proc`.  In fact, I kind of like the idea that the user can use the `frozen` state of the default object to control whether or not Grape should expend resources cloning it each time it's used.

When the default value is _not_ frozen, I agree that `dup` is probably preferable to `clone`.